### PR TITLE
Dotenv support

### PR DESCRIPTION
**Pull request by sokokaleb**
Dotenv support is provided by @vlucas's [phpdotenv](https://github.com/vlucas/phpdotenv).

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ENCRYPTION_KEY=
+DB_HOSTNAME=
+DB_USERNAME=
+DB_PASSWORD=
+DB_NAME=
+CPP_EXECUTABLE_PATH=/usr/bin/g++
```
Comment (fushar) [inline .env.example]: Please fix your editor's setting to automatically add newline at the end of files

File: README.md [hunk 2]
```diff
@@ -42,26 +42,21 @@ Download a copy of the latest Regrader: https://github.com/fushar/regrader/relea
...
 ```
 
-It is advisable that the string consists of 32 random characters.
+And edit each entry of `.env` based on your system configuration:
+
+- `ENCRYPTION_KEY` denotes the encryption key that will be used by CodeIgniter. It is advisable that the string consists of 32 random characters.
+- `DB_HOSTNAME` denotes the hostname of the MySQL database.
+- `DB_USERNAME` and `DB_PASSWORD` denote the credentials of the MySQL user related to the database.
+- `DB_NAME` denotes the database name that will be used by Regrader.
+- `CPP_EXECUTABLE_PATH` denotes the C++ compiler path that will be used to compile checker source code (e.g., `usr/bin/g++`).
```
Comment (fushar) [inline README.md]: Only for checker? How about the grader? What if I don't use checker at all?

Comment (fushar) [inline README.md]: `/usr/bin/g++`

Comment (fushar) [inline README.md]: Why not `CPP_COMPILER_PATH` ?

File: .env.example (hunk 1 above)
Comment (fushar) [inline .env.example]: Give sensible example values for encryption keys, hostname, etc. as well

File: README.md [hunk 3]
```diff
@@ -42,26 +42,21 @@ Download a copy of the latest Regrader: https://github.com/fushar/regrader/relea
...
-$config['encryption_key'] = '';
+```bash
+$ cp .env.example .env
 ```
 
-It is advisable that the string consists of 32 random characters.
+And edit each entry of `.env` based on your system configuration:
+
+- `ENCRYPTION_KEY` denotes the encryption key that will be used by CodeIgniter. It is advisable that the string consists of 32 random characters.
+- `DB_HOSTNAME` denotes the hostname of the MySQL database.
```
Comment (fushar) [inline README.md]: denotes the hostname of the database server.

File: README.md [hunk 4]
```diff
@@ -42,26 +42,21 @@ Download a copy of the latest Regrader: https://github.com/fushar/regrader/relea
...
+- `DB_USERNAME` and `DB_PASSWORD` denote the credentials of the MySQL user related to the database.
```
Comment (fushar) [inline README.md]: denotes the user credentials of the database.

**Comment by fushar** [comment]
Please fix the inline comments

File: README.md (hunk 2 above)
Comment (sokokaleb) [inline README.md]: This is only for compiling checker source code.
The compiler path used in grading would be editable via Web UI.

Comment (fushar) [inline README.md]: Right, I completely forgot about that :)

Just give a note, something along the line ("... if you don't wish to use checker at all, leave it empty")

**Comment by sokokaleb** [comment]
@fushar fixes added

**Comment by fushar** [comment]
Squashed & merged as 723fa71c8b042e9228d6e00cacb44d9bd287daf8, thanks!

**Pull request closed without merging**